### PR TITLE
Fix issues related to 1.3.0 release

### PR DESCRIPTION
**Pull request by AdilHoumadi**
…ed implement FactoryInterface
- [FIX] Fix mapping for (DbRecordExistsFactory and DbNoRecordExistsFactory)
- [FIX] Fix documentation for the mapping part

/cc @weierophinney 
Thanks

### CODE REVIEW

**Comment by MichaelGooden** [comment]
@weierophinney These bugs actually break projects by throwing PHP Fatal errors. Please look into this ASAP.

**Comment by MichaelGooden** [comment]
Ping @ezimuel, please could you have a look at this, mwop is on holiday this week and these bugs straight out break projects who update to this version. Thanks!

**Comment by webimpress** [comment]
It looks like they both are on holiday:
> I'm on vacation next week, as is Enrico.
https://github.com/zfcampus/zf-apigility/issues/167#issuecomment-232736596


As temporary solution I'd suggest to add restrictions in your `composer.json`:
```
zfcampus/zf-content-validation: "1.2.0"
```

and:
```
zfcampus/zf-mvc-auth: "1.3.2"
```
if issue https://github.com/zfcampus/zf-mvc-auth/pull/120 is affecting you as well.

**Comment by jguittard** [comment]
I'll look into these two today.

**Comment by AdilHoumadi** [comment]
Hi @akrabat, can you check other fixes in this PR ? 
Thanks

**Comment by MichaelGooden** [comment]
@AdilHoumadi We are working on this in IRC, please stand by.

**Comment by AdilHoumadi** [comment]
@MichaelGooden, thanks ! 
can you share the channel name ?

**Comment by MichaelGooden** [comment]
@AdilHoumadi #zftalk.dev on freenode

**Comment by AdilHoumadi** [comment]
@MichaelGooden, Thanks!

File: config/module.config.php [hunk 1]
```diff
@@ -43,8 +43,8 @@
     ],
     'validators' => [
         'factories' => [
-            Validator\DbRecordExists::class => Validator\Db\RecordExistsFactory::class,
-            Validator\DbNoRecordExists::class => Validator\Db\NoRecordExistsFactory::class,
+            'ZF\ContentValidation\Validator\DbRecordExists' => Validator\Db\RecordExistsFactory::class,
+            'ZF\ContentValidation\Validator\DbNoRecordExists' => Validator\Db\NoRecordExistsFactory::class,
```
Comment (weierophinney) [inline config/module.config.php]: Just FYI: PHP will resolve the `::class`-notated strings just fine, even if the class does not exist. Essentially, it checks against existing imports and namespace declarations in order to resolve to a name, but does not check to see if that name exists. As such, the above was never an issue.

File: src/ContentValidationListenerFactory.php [hunk 2]
```diff
@@ -10,12 +10,14 @@
 use Zend\ServiceManager\FactoryInterface;
 use Zend\ServiceManager\ServiceLocatorInterface;
 
-class ContentValidationListenerFactory
+class ContentValidationListenerFactory implements FactoryInterface
```
Comment (weierophinney) [inline src/ContentValidationListenerFactory.php]: This was not necessary, either; the original was done as a plain invokable.

**Comment by AdilHoumadi** [comment]
Thanks for you feedback, we noticed that! you can check this thread : https://github.com/zfcampus/zf-content-validation/pull/76#discussion_r71698040

File: src/ContentValidationListenerFactory.php (hunk 2 above)
Comment (AdilHoumadi) [inline src/ContentValidationListenerFactory.php]: Thanks for you feedback, we noticed that! you can check this thread : https://github.com/zfcampus/zf-content-validation/pull/76#discussion_r71698040

File: config/module.config.php (hunk 1 above)
Comment (AdilHoumadi) [inline config/module.config.php]: Thanks for you feedback, we noticed that! you can check this thread : https://github.com/zfcampus/zf-content-validation/pull/76#discussion_r71698040

File: src/ContentValidationListenerFactory.php (hunk 2 above)
Comment (AdilHoumadi) [inline src/ContentValidationListenerFactory.php]: Without this implementation we get the error mentioned here : https://github.com/zfcampus/zf-content-validation/issues/72

Comment (weierophinney) [inline src/ContentValidationListenerFactory.php]: The other option would have been to remove the second and third arguments. However, I'm fine with the fix; I forgot I'd kept `createService()` for BC purposes.

**Pull request closed without merging**